src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -3412,6 +3425,30 @@ internal unsafe bool IsActualEnum
             }
         }
 
+        internal new unsafe bool IsInterface
+        {
+            get
+            {
+                TypeHandle th = GetNativeTypeHandle();
+
+                bool isInterface = !th.IsTypeDesc && th.AsMethodTable()->IsInterface;
+                GC.KeepAlive(this);
```

Comment:
I wonder if JIT is allowed to fold `typeof(T).IsInterface` without intrinsics at all - the RuntimeType object is on the NonGC heap, `GetNativeTypeHandle()` is basically MethodTable which is mostly immutable 🙂

Reply:
Yeah, it would be possible in theory. I am not sure whether it would make things simpler overall.